code/datums/components/thermite.dm

```diff
@@ -0,0 +1,70 @@
+/datum/component/thermite
+	dupe_mode = COMPONENT_DUPE_UNIQUE
+	var/static/list/blacklist = typecacheof(/turf/closed/wall/mineral/diamond)
+	var/static/list/resistlist = typecacheof(/turf/closed/wall/r_wall)
+	var/amount
+	var/overlay
+
+/datum/component/thermite/Initialize(_amount)
+	. = ..()
+
+	if(!istype(parent, /turf))
+		qdel(src)
```

Comment:
QDEL_IN(src, 0)

Reply:
We agreed that just straight qdel() is fine when the component has been applied to something incorrectly.